src/modules/map/components/layerManager/LayerManager.js

```diff
@@ -0,0 +1,156 @@
+			const displayProperties = {
+				collapsed:true,
+				visible:true
+			};
+			if(old) {
+				displayProperties.collapsed = old.collapsed;
+				displayProperties.visible = old.visible;
+			}
+			draggableItems.push({ ...layer, isPlaceholder:false, listIndex:j + 1, isDraggable:true, ...displayProperties });
+			draggableItems.push({ zIndex:layer.zIndex + 1, isPlaceholder:true, listIndex:j + 2, isDraggable:false });
+			j += 2;
+		}		
+		this._draggableItems = draggableItems;		
+	}
+
+	/**
+	 * @override
+	 */
+	createView() {
+		const translate = (key) => this._translationService.translate(key);
+		const { active } = this._state;
+		const layerCount = active.length;
+		this._buildDraggableItems(active);
+
+		const isNeighbour = (index, otherIndex) => {
+			return index === otherIndex || index - 1 === otherIndex || index + 1 === otherIndex;
+		};		
+
+		const createLayerElement = (layerItem) => {		
+
+			return html`<ba-layer-item .layer=${layerItem} class='layer' draggable>
+					</ba-layer-item>`;
+			
+		};
+
+		const createPlaceholderElement = (layerItem) => {
+			return html`<div id=${'placeholder_' + layerItem.listIndex} class='placeholder'>							
+						</div>`;
+		};
+
+		const onDragStart = (e, layerItem) => {
+			this._draggedItem = layerItem;
+			e.dataTransfer.dropEffect = 'move';
+			e.dataTransfer.effectAllowed = 'move';		
+		};
+
+		const onDragEnd = (e) => {
+			e.preventDefault();
+		};
+
+		const onDrop = (e, layerItem) => {
+			if(layerItem.isPlaceholder && this._draggedItem) {
+				let newZIndex = layerItem.zIndex;
+				if(layerItem.zIndex === this._layerCount - 1) {
+					newZIndex = layerItem.zIndex - 1;
+				}
+				modifyLayer(this._draggedItem.id, { zIndex:newZIndex });				
+			}
+			if(e.target.classList.contains('placeholder')) {
+				e.target.classList.remove('over');				
+			}
+			this._resetDraggedItem();
+		};
+		const onDragOver = (e, layerItem) => {
+			e.preventDefault();			
+			let dropEffect = 'none';
+			
+			if(this._draggedItem) {			
+				if(layerItem.isPlaceholder && !isNeighbour(layerItem.listIndex, this._draggedItem.listIndex)) {
+					dropEffect = 'all';
+				}
+			}
+			e.dataTransfer.dropEffect = dropEffect;
+		};
+
+		const onDragEnter = (e, layerItem) => {
+			if(this._draggedItem) {			
+				if(layerItem.isPlaceholder && !isNeighbour(layerItem.listIndex, this._draggedItem.listIndex)) {
+					e.target.classList.add('over');
+				}
+			}
+		};
+		const onDragLeave = (e) => {			
+			e.stopPropagation();
+			if(e.target) {
+				if(e.target.classList.contains('over')) {
+					e.target.classList.remove('over');					
+				}			
+			}			
+		};
+
+		return html`
+			<style>${css}</style>
+			<div class="layermanager overflow-container">
+				<div class='title'>${translate('layer_manager_title')} (${layerCount})</div> 
+				<ul class='layers'>
+                    ${repeat(this._draggableItems, (layerItem) => layerItem.listIndex, (layerItem, index) => html`
```

Comment:
Here we have still a problem: the id for the item is not unique, that can can cause some confusion during the rendering phase. (from the docu: _keyFunction is a function that takes a single item as an argument and returns a guaranteed unique key for that item._)
Due to this,  you have to call `render()` manually in `LayerItem.js` line 121, which wouldn't be necessary otherwise. 
But that's just the effect, the problem is, that the repeater cannot reuse your items correctly

Reply:
Ah...ok...i fix this. I did not realize this as update-render-mechanism, only as an index for a sortable

Reply:
Yes, based on a stable and unique item id, the reapter "knows" which which part of its items have to be re-rerndered.